Add add.uw instruction

Vokerlee opened the pull request:
Add unsigned word instruction (RISC-V Bit-Manipulation ISA-extensions).
Opcode : 1101110.

This instruction performs an XLEN-wide addition between rs2
and the zero-extended least-significant word of rs1:

let base = X(rs2);
let index = EXTZ(X(rs1)[31..0]);
X(rd) = base + index;

pavelkryukov commented on simulator/risc_v/t/unit_test.cpp:
```diff
@@ -353,6 +354,9 @@ TEST_RV32_RR_OP( 3, minu, 0x22222222, 0xbbbbbbbb, 0x22222222)
 TEST_RV64_RR_OP( 1, packu, 0x11111111ffffffff, 0xffffffff22222222, 0x1111111133333333)
 
+TEST_RV64_RR_OP( 1, add_uw, 0x1111111155555555, 0x5555555522222222, 0x1111111133333333)
+TEST_RV64_RR_OP( 2, add_uw, 0x12344321b5a69788, 0xabababab82736455, 0x1234432133333333)
```
This test case is just obfuscated version of the test case above. Not sure if we need it.

pavelkryukov replied:
Could you please test overflow cases (what if 32 bit word overflows, what if 64 bit overflows)?

Vokerlee committed:
Add add.uw instruction

Add unsigned word instruction (RISC-V Bit-Manipulation ISA-extensions).
Opcode : 1101110.

This instruction performs an XLEN-wide addition between rs2
and the zero-extended least-significant word of rs1:

let base = X(rs2);
let index = EXTZ(X(rs1)[31..0]);
X(rd) = base + index;
